From 9caf8c430937a3308cffc8ff56766dc3c2512129 Mon Sep 17 00:00:00 2001
From: Nicola Fontana <ntd@entidi.it>
Date: Mon, 23 Mar 2015 22:04:32 +0100
Subject: [PATCH 10/10] Use a launcher to execute g-ir-scanner

In cross-compilation the build system is unable to execute the compiled
binary. Give the possibility to use  a launcher, e.g. wine for MinGW or
qemu for different CPU type.
---
 common.mk                | 7 ++++++-
 giscanner/gdumpparser.py | 6 ++++++
 2 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/common.mk b/common.mk
index 06d22d4..e54f285 100644
--- a/common.mk
+++ b/common.mk
@@ -4,7 +4,11 @@
 # This file defines variables that are compatible with
 # Makefile.introspection, but for use within the gobject-introspection
 # module itself.
-#
+
+
+# The command to use for executing g-ir-compiler. Normally should be
+# empty but can be defined (e.g. wine or qemu) during cross-compilation
+INTROSPECTION_LAUNCHER =
 
 INTROSPECTION_SCANNER = \
     env PATH=".libs:$(PATH)" \
@@ -25,6 +29,7 @@ INTROSPECTION_SCANNER_ARGS = \
 
 INTROSPECTION_COMPILER = \
     env PATH=".libs:$(PATH)" \
+        $(INTROSPECTION_LAUNCHER) \
         $(top_builddir)/g-ir-compiler$(EXEEXT)
 
 INTROSPECTION_COMPILER_ARGS = \
diff --git a/giscanner/gdumpparser.py b/giscanner/gdumpparser.py
index e1fc935..1ea5981 100644
--- a/giscanner/gdumpparser.py
+++ b/giscanner/gdumpparser.py
@@ -158,6 +158,12 @@ blob containing data gleaned from GObject's primitive introspection."""
         out_path = os.path.join(self._binary.tmpdir, 'dump.xml')
 
         args = []
+
+        # Prepend the emulator command and arguments, if defined
+        emulator = os.environ.get('INTROSPECTION_LAUNCHER')
+        if emulator:
+            args.extend(emulator.split())
+
         args.extend(self._binary.args)
         args.append('--introspect-dump=%s,%s' % (in_path, out_path))
 
-- 
2.3.3

